WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/config/ColorSelectionActivity.kt

```diff
@@ -15,51 +15,77 @@
  */
 package com.example.android.wearable.watchfacekotlin.config
 
-import android.app.Activity
+import android.content.Context
+import android.content.SharedPreferences
 import android.os.Bundle
-import android.view.View
-
-import androidx.recyclerview.widget.LinearLayoutManager
-import androidx.wear.widget.WearableRecyclerView
-
+import androidx.activity.ComponentActivity
+import androidx.core.content.edit
 import com.example.android.wearable.watchfacekotlin.R
-import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData
+import com.example.android.wearable.watchfacekotlin.databinding.ActivityColorSelectionConfigBinding
+import com.example.android.wearable.watchfacekotlin.watchface.AnalogWatchFace
+import java.util.ArrayList
 
 /**
- * Allows user to select color for something on the watch face (background, highlight,etc.) and
- * saves it to [android.content.SharedPreferences] in RecyclerView.Adapter.
+ * Allows user to select a color for component on the watch face (background, highlight, etc.) and
+ * saves it to [android.content.SharedPreferences] for later use.
  */
-class ColorSelectionActivity : Activity() {
+class ColorSelectionActivity : ComponentActivity() {
 
-    private lateinit var configAppearanceWearableRecyclerView: WearableRecyclerView
+    private lateinit var binding: ActivityColorSelectionConfigBinding
     private lateinit var colorSelectionRecyclerViewAdapter: ColorSelectionRecyclerViewAdapter
 
+    private lateinit var sharedPref: SharedPreferences
+    private lateinit var sharedPrefString: String
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_color_selection_config)
 
-        // Assigns SharedPreference String used to save color selected.
-        val sharedPrefString = intent.getStringExtra(EXTRA_SHARED_PREF)
+        sharedPref = getSharedPreferences(
+            AnalogWatchFace.analog_complication_preference_file_key,
+            Context.MODE_PRIVATE)
```

Comment:
Can this be accessed off the main thread (using coroutines) to avoid janking the UI with blocking I/O?

Reply:
I haven't see this use with SharedPreferences. [The guide](https://developer.android.com/training/data-storage/shared-preferences#ReadSharedPreference) calls out using this method as is, and I couldn't find any information to use coroutines in that manner. The guide does say to use `apply()` for async edits (which I do now). 

Can you give me an example to research?

I might leave "as is" if not. I plan to eventually move this to a Room database to match my alpha sample in the future.

Reply:
You should not add the `apply()` call inside the edit block of the shared prefs. It will apply automatically. See https://developer.android.com/kotlin/ktx (first sample)

Reply:
Sorry, that shouldn't have gotten back in. Fixed. Thanks!